panel/io/embed.py

```diff
@@ -258,22 +260,25 @@ def embed_state(panel, model, doc, max_states=1000, max_opts=3,
     cross_product = list(product(*[vals[::-1] for _, _, vals, _ in values]))
 
     if len(cross_product) > max_states:
-        raise RuntimeError('The cross product of different application '
-                           'states is too large to explore (N=%d), either reduce '
-                           'the number of options on the widgets or increase '
-                           'the max_states specified on static export.' %
+        param.main.warning('The cross product of different application '
+                           'states is very large to explore (N=%d), consider '
+                           'reducing the number of options on the widgets or '
+                           'increase the max_states specified in the function '
+                           'to remove this warning' %
                            len(cross_product))
 
     nested_dict = lambda: defaultdict(nested_dict)
     state_dict = nested_dict()
     changes = False
-    for key in cross_product:
+    for key in tqdm(cross_product):
         sub_dict = state_dict
         skip = False
         for i, k in enumerate(key):
             w, m, _, g = values[i]
             try:
                 w.value = k
+            except (KeyboardInterrupt, SystemExit):
+                raise
             except:
```

Comment:
This should be replaced with simpler and idiomatic `except Exception`, which filers out system-type exceptions.

Reply:
The keyboard expedition was added to allow Control C breaking a long process- it was tested to work. Without this code, one could not stop a long process running. Is Exception setup to handle those interrupts? 
If yes, I can test it.

Reply:
They way this works is that bare `except` is equivalent to `except BaseException`, which is a catch-all clause. System-type exceptions (interrupt, exit, etc.) derive directly from it. There is also `Exception` to cover all other (user) exceptions. See Python's [exception hierarchy](https://docs.python.org/3/library/exceptions.html#exception-hierarchy) for details. So, if you change this to `except Exception` is will catch only user exceptions and allow to pass-though system exceptions. In general it's a bad practice in Python to use bare `except`, because the side effect is that Ctrl+C, etc. don't work as expected. In fact panel uses those a lot (some 50 cases), so I will go ahead and fix that and set up linting to cover this issue.

Reply:
Thanks Mateusz, 

You explained well this time - you meant that instead of isolating the keyboard error, just avoid using a bare except - this makes sense. I was looking at the code you highlighted rather than the code below. If I understand you correctly, you want the code to look like:

````

        for i, k in enumerate(key):
            w, m, _, g = values[i]
            try:
                w.value = k
            except Exception:
                skip = True
                break
````
I just fixed one problem, not the code everywhere as you are suggesting.

Hopefully we are on the same page now. 

It yes, will be happy to fix it.